src/commands/appdistribution-distribute.ts

```diff
@@ -210,56 +224,78 @@ export const command = new Command("appdistribution:distribute <release-binary-f
     await requests.distribute(releaseName, testers, groups);
 
     // Run automated tests
-    if (testDevices?.length) {
-      utils.logBullet("starting automated tests (note: this feature is in beta)");
-      const releaseTest = await requests.createReleaseTest(
-        releaseName,
-        testDevices,
-        loginCredential,
-      );
-      utils.logSuccess(`Release test created successfully`);
+    if (testDevices.length) {
+      utils.logBullet("starting automated test (note: this feature is in beta)");
+      const releaseTestPromises: Promise<ReleaseTest>[] = [];
+      if (!testCases.length) {
+        // fallback to basic automated test
+        releaseTestPromises.push(
+          requests.createReleaseTest(releaseName, testDevices, loginCredential),
+        );
+      } else {
+        for (const testCaseId of testCases) {
+          releaseTestPromises.push(
+            requests.createReleaseTest(
+              releaseName,
+              testDevices,
+              loginCredential,
+              `${appName}/testCases/${testCaseId}`,
+            ),
+          );
+        }
+      }
+      const releaseTests = await Promise.all(releaseTestPromises);
+      utils.logSuccess(`${releaseTests.length} release test(s) started successfully`);
       if (!options.testNonBlocking) {
-        await awaitTestResults(releaseTest.name!, requests);
+        const releaseTestNames = new Set(releaseTests.map((rt) => rt.name!!));
```

Comment:
Unfortunately `?? ""` or something similar doesn't work, since there's no reasonable default that could be used. If the `ReleaseTest` resource returned by the API doesn't have the `name` property set, then the subsequent code can't work.

I could add an explicit check and error message, but that seems to be overkill. There are many other edge-cases which are much more deserving explicit error checking. This one here can't really happen - unless the API is completely b0rken.

I've replaced the `!!` with a single `!`